tests/unit_tests/serialization.cpp

```diff
@@ -616,6 +616,48 @@ TEST(Serialization, serializes_ringct_types)
   ASSERT_EQ(bp0, bp1);
 }
 
+TEST(Serialization, key_encryption_transition)
+{
+  const cryptonote::network_type nettype = cryptonote::TESTNET;
+  tools::wallet2 w(nettype);
+  const boost::filesystem::path wallet_file = unit_test::data_dir / "wallet_9svHk1";
+  const boost::filesystem::path key_file = unit_test::data_dir / "wallet_9svHk1.keys";
+  const boost::filesystem::path backup_wallet_file = unit_test::data_dir / "wallet_9svHk1_backup";
+  const boost::filesystem::path backup_key_file = unit_test::data_dir / "wallet_9svHk1_backup.keys";
+  string password = "test";
+  bool r = false;
+
+  // Copy the original files before we overwrite them for this test
+  boost::filesystem::copy(wallet_file,backup_wallet_file);
+  boost::filesystem::copy(key_file,backup_key_file);
+
+  try
+  {
+    // Key transition
+    w.load(wallet_file.string(), password); // legacy decryption method
+    ASSERT_TRUE(w.get_load_info().is_legacy_key_encryption);
+    const crypto::secret_key view_secret_key = w.get_account().get_keys().m_view_secret_key;
+
+    w.rewrite(wallet_file.string(), password); // transition to new key format
+
+    w.load(wallet_file.string(), password); // new decryption method
+    ASSERT_FALSE(w.get_load_info().is_legacy_key_encryption);
+    ASSERT_EQ(w.get_account().get_keys().m_view_secret_key,view_secret_key);
+
+    r = true;
+  }
+  catch (const exception& e)
+  {}
+
+  // Restore the original files
+  MINFO("Restoring wallet file...");
+  boost::filesystem::rename(backup_wallet_file,wallet_file);
+  MINFO("Restoring key file...");
+  boost::filesystem::rename(backup_key_file,key_file);
+
+  ASSERT_TRUE(r);
+}
```

Comment:
It would seem better to do the test on the scratch file, rather than the revision controlled one (ie, if the process SIGSEGVs for instance, you end up with a git change).

Reply:
Updated.